### PR TITLE
Feature/cc 0058 a user can view folder

### PR DESCRIPTION
**Pull request by Beryl-Zhou**

### CODE REVIEW

File: src/components/EditorSideBar/EditorSideBarContent.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Box } from "@chakra-ui/react";
 import React from "react";
 import { ChevronLeftIcon } from "@chakra-ui/icons";
 import Blueprints from "./Blueprints";
+import Folder from "./Folder";
```
Comment (stevenliu986) [inline src/components/EditorSideBar/EditorSideBarContent.tsx:5]: Is this "Folder" on SideBar? If so, should the path be /components/EditorSideBar/Folder?

File: src/components/NavBar/EditorDesignName.tsx [hunk 2]
```diff
@@ -8,15 +7,17 @@ import {
   Input,
   EditableInput,
 } from "@chakra-ui/react";
-import { useState } from "react";
-import { BiStar, BiPencil } from "react-icons/bi";
+import { BiPencil } from "react-icons/bi";
 import { useDispatch } from "react-redux";
 import { useStoreSelector } from "@/store/hooks";
+import { changeDesignName } from "@/store/reducer/courtSizeSlice";
```
Comment (stevenliu986) [inline src/components/NavBar/EditorDesignName.tsx:13]: We no longer use courtSizeSlice reducer. We use courtSpecDataSlice instead. So, please move this reducer to courtSpecDataSlice.

File: src/components/TopBar/SaveBoard.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useState } from "react";
+import { Box, Flex, Input, Text } from "@chakra-ui/react";
+import DocSvg from "@/assets/svg/TopBarSvg/document.svg";
+import SaveSvg from "@/assets/svg/TopBarSvg/save.svg";
+import { useRef } from "react";
```
Comment (stevenliu986) [inline src/components/TopBar/SaveBoard.tsx:5]: Why not import this hook in the first line?

File: src/components/TopBar/SaveBoard.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import { AlertDialogCloseButton, Button, useDisclosure } from "@chakra-ui/react";
```
Comment (stevenliu986) [inline src/components/TopBar/SaveBoard.tsx:6]: Why not import this hook in the 2nd line?

File: src/components/TopBar/SaveBoard.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogContent,
+  AlertDialogOverlay,
+} from "@chakra-ui/react";
```
Comment (stevenliu986) [inline src/components/TopBar/SaveBoard.tsx:14]: same as line 6 comment.

File: src/store/reducer/courtSizeSlice.ts [hunk 6]
```diff
@@ -52,7 +57,7 @@ export const courtWhiteLine = initialState.strokeWidth / 3;
 export const dashedWhiteLine = initialState.strokeWidth / 5;
 // export const borderLength = initialState.borderLength;
 
-export const { changeCourtSize, changeBorderLength } = courtSizeSlice.actions;
+export const { changeCourtSize, changeBorderLength, changeDesignName } = courtSizeSlice.actions;
 export const courtSizeData = (state: RootState) => state.courtSize;
 
 export default courtSizeSlice.reducer;
```
Comment (stevenliu986) [inline src/store/reducer/courtSizeSlice.ts:63]: courtSizeSlice is deprecated.

**Review by stevenliu986 — COMMENTED** [review]
May need some code modifications.

File: src/__tests__/components/TopBar/index.test.tsx [hunk 7]
```diff
@@ -32,18 +32,18 @@ describe("TopBar", () => {
...
+  //   // TODO: Add this test after finishing putting user login status into redux
+  //   // const loginModalDialog = screen.getByRole("dialog");
+  //   // await waitFor(() => expect(loginModalDialog).toBeVisible());
   // });
+
+  // // it("Should dispatch the action", () => {
+  // //   const dispatchAction = jest.spyOn(redux, "useDispatch");
+  // //   render(<TopBar />);
+  // //   expect(dispatchAction).toHaveBeenCalled();
+  // // });
```
Comment (Jzeva) [inline src/__tests__/components/TopBar/index.test.tsx:48]: Maybe we can delete unused code.

**Review by Jzeva — APPROVED** [review]
The save function works for me. Just don't forget to make it only available after users login in the future.

**Review by GitHubEason — APPROVED** [review]

**Pull request merged**